src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1704,9 +1705,13 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
                         break;
                     case $usedEnvs || preg_match('#^[a-z]++://#', $storeDsn):
                         if (!$container->hasDefinition($connectionDefinitionId = $container->hash($storeDsn))) {
+                            if (!class_exists(ConnectionFactory::class)) {
+                                throw new LogicException('Lock stores cannot be configured as the Dsn component is not installed.');
```

Comment:
should be removed once dsn is added as a requirement in composer.json

Reply:
removed (same for CachePoolPass)